lib/pharos/phase_manager.rb

```diff
@@ -29,32 +30,20 @@ def initialize(**options)
 
     # @param phases [Array<Pharos::Phases::Base>]
     # @return [Array<...>]
-    def run_parallel(phases, &block)
-      threads = phases.map { |phase|
-        Thread.new do
-          Thread.current.report_on_exception = false
-          Retry.perform(yield_object: phase, logger: logger, exceptions: RETRY_ERRORS, &block)
-        end
-      }
-      threads.map(&:value)
-    end
-
-    # @param phases [Array<Pharos::Phases::Base>]
-    # @return [Array<...>]
-    def run_serial(phases, &block)
-      phases.map do |phase|
-        Retry.perform(yield_object: phase, logger: logger, exceptions: RETRY_ERRORS, &block)
-      end
-    end
-
-    # @param phases [Array<Pharos::Phases::Base>]
-    # @param parallel [Boolean]
-    # @return [Array<...>]
-    def run(phases, parallel: true, &block)
-      if parallel
-        run_parallel(phases, &block)
-      else
-        run_serial(phases, &block)
+    def run(phases, &block)
+      begin
+        threads = phases.map { |phase|
+          Thread.new do
+            Thread.current.report_on_exception = false
+            Thread.current.abort_on_exception = true
+            Retry.perform(yield_object: phase, logger: logger, exceptions: RETRY_ERRORS, &block)
+          end
+        }
+        threads.map(&:value)
+      rescue StandardError
+        threads.map(&:kill)
+        sleep 0.5 until threads.all?(&:stop?)
```

Comment:
Without this the threads often output some lines after the exception is printed.